.buildkite/pipeline.yml

```diff
@@ -196,7 +196,8 @@ steps:
     artifact_paths:
       - "build/TEST-go-unit.cov"
     agents:
-      image: "golang:1.22.6"
+      # NOTE: use the go-version env variable that's set in the pre-command hook
+      image: "golang:${GO_VERSION}"
```

Comment:
I don't think variables in the precommand hook get evaluated here (only after an agent has started running in the context of a shell)

Reply:
That's indeed an excellent point!

I guess I'll need to use updatecli for this file then